### DISCUSSION TITLE
Specify source UDP port

### DISCUSSION THREAD

**Discussion by sirasjad** [jtpereyda/boofuzz#574] [Q&A] [answered]
Hi, I'm trying to fuzz a target over UDP and I'm not sure how to define the source port. The fuzzed network packets must be sent from a specific source port, otherwise the target will ignore these packets. 

The target IP is 192.168.1.50 (port 1337) and the source (sender) port must be 1234. I have tried binding ("0.0.0.0", 1234) and ("127.0.0.1", 1234) but it doesn't seem to work. 

```
session = Session(target=Target(connection=UDPSocketConnection("192.168.1.50", 1337, bind=("0.0.0.0", 1234))))
```

How can I achieve this? Thanks in advance!

**Comment by SR4ven** [comment, marked as answer]
Hmm I'm no expert at UDP but that's how I'd give it a try. They say the same [here](https://stackoverflow.com/questions/2694212/python-set-source-port-number-with-sockets).
Looking at our code, that's exactly what should happen when executing your example.

You could try `UDPSocketConnection("192.168.1.50", 1337, bind=("", 1234)`, maybe that works.

Another option would be to assemble your own UDP header using the raw socket connection classes, but that is of course more work.

**Comment by sirasjad** [comment]
Turns out I had to use bind() after all. I had some OS issues, it's solved now.
